src/main/kotlin/org/rust/lang/core/resolve/ImplLookup.kt

```diff
@@ -199,27 +197,14 @@ data class ParamEnv(val callerBounds: List<TraitRef>) {
 
 class ImplLookup(
     private val project: Project,
-    cargoProject: CargoProject?,
+    private val containingCrate: Crate?,
     val items: KnownItems,
     private val paramEnv: ParamEnv = ParamEnv.EMPTY
 ) {
     // Non-concurrent HashMap and lazy(NONE) are safe here because this class isn't shared between threads
-    private val traitSelectionCache: Cache<TraitRef, SelectionResult<SelectionCandidate>> =
-        if (paramEnv.isEmpty() && cargoProject != null) {
-            cargoProjectGlobalTraitSelectionCache.getCache(cargoProject)
-        } else {
-            // function-local cache is used when [paramEnv] is not empty, i.e. if there are trait bounds
-            // that affect trait selection
-            Cache.new()
-        }
-    private val findImplsAndTraitsCache: Cache<Ty, List<TraitImplSource>> =
-        if (cargoProject != null) {
-            cargoProjectGlobalFindImplsAndTraitsCache.getCache(cargoProject)
```

Comment:
`cargoProjectGlobalFindImplsAndTraitsCache` and `CargoProjectCache` class can be removed now